sklearn/neural_network/tests/test_mlp.py

```diff
@@ -556,3 +562,29 @@ def test_adaptive_learning_rate():
     clf.fit(X, y)
     assert_greater(clf.max_iter, clf.n_iter_)
     assert_greater(1e-6, clf._optimizer.learning_rate)
+
+
+def test_warm_start():
+    X = X_iris
+    y = y_iris
+
+    y_2classes = np.array([0] * 75 + [1] * 75)
+    y_3classes = np.array([0] * 40 + [1] * 40 + [2] * 70)
+    y_3classes_alt = np.array([0] * 50 + [1] * 50 + [3] * 50)
+    y_4classes = np.array([0] * 37 + [1] * 37 + [2] * 38 + [3] * 38)
+    y_5classes = np.array([0] * 30 + [1] * 30 + [2] * 30 + [3] * 30 + [4] * 30)
+
+    with ignore_warnings(category=Warning):
```

Comment:
It was mentioned in a comment above: "Sometimes it throws a RuntimeWarning: underflow encountered in exp np.exp(tmp, out=X). I tried ignore_warnings(category=RuntimeWarning) but it didn't seem to work." 

I also tried to use np.errstate but couldn't get it to work either.